src/atomic.cr

```diff
@@ -410,13 +464,13 @@ end
 # ```
 struct Atomic::Flag
   def initialize
-    @value = 0_u8
+    @value = Atomic(Bool).new(false)
   end
 
   # Atomically tries to set the flag. Only succeeds and returns `true` if the
   # flag wasn't previously set; returns `false` otherwise.
   def test_and_set : Bool
-    ret = Atomic::Ops.atomicrmw(:xchg, pointerof(@value), 1_u8, :sequentially_consistent, false) == 0_u8
+    ret = @value.swap(true, :sequentially_consistent) == false
     {% if flag?(:arm) %}
       Atomic::Ops.fence(:sequentially_consistent, false) if ret
     {% end %}
```

Comment:
@Sija Good question. I'm thinking more and more that if we define a memory order on an atomic operation and the atomics on weak architectures don't, then we should add the barrier right into the barrier (unless the ordering is relaxed).

It might be inefficient in some cases, but you might want to specialize your algorithm for weak architectures, instead of forcing every usage to have a special case for ARM (and not support other weak architectures if we eventually add them).